test/python/pulse/test_builder.py

```diff
@@ -0,0 +1,950 @@
+
+        self.assertEqual(schedule, reference)
+
+    def test_call_circuit(self):
+        """Test calling circuit instruction."""
+        inst_map = self.inst_map
+        reference = inst_map.get('u1', (0,), 0.0)
+
+        u1_qc = circuit.QuantumCircuit(2)
+        u1_qc.u1(0.0, 0)
+
+        transpiler_settings = {'optimization_level': 0}
+
+        with pulse.build(self.backend,
+                         default_transpiler_settings=transpiler_settings
+                         ) as schedule:
+            with pulse.align_right():
+                pulse.call_circuit(u1_qc)
+
+        self.assertEqual(schedule, reference)
+
+
+class TestDirectives(TestBuilder):
+    """Test context builder directives."""
+
+    def test_barrier_with_align_right(self):
+        """Test barrier directive with right alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as schedule:
+            with pulse.align_right():
+                pulse.delay(3, d0)
+                pulse.barrier(d0, d1, d2)
+                pulse.delay(11, d2)
+                with pulse.align_right():
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        schedule = transforms.remove_directives(schedule)
+
+        reference = pulse.Schedule()
+        # d0
+        reference = reference.insert(0, instructions.Delay(3, d0))
+        reference = reference.insert(7, instructions.Delay(7, d0))
+        # d1
+        reference = reference.insert(9, instructions.Delay(5, d1))
+        # d2
+        reference = reference.insert(3, instructions.Delay(11, d2))
+
+        self.assertEqual(schedule, reference)
+
+    def test_barrier_with_align_left(self):
+        """Test barrier directive with left alignment context."""
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+        d2 = pulse.DriveChannel(2)
+
+        with pulse.build() as schedule:
+            with pulse.align_left():
+                pulse.delay(3, d0)
+                pulse.barrier(d0, d1, d2)
+                pulse.delay(11, d2)
+                with pulse.align_left():
+                    pulse.delay(5, d1)
+                    pulse.delay(7, d0)
+
+        schedule = transforms.remove_directives(schedule)
+
+        reference = pulse.Schedule()
+        # d0
+        reference += instructions.Delay(3, d0)
+        reference += instructions.Delay(7, d0)
+        # d1
+        reference = reference.insert(3, instructions.Delay(5, d1))
+        # d2
+        reference = reference.insert(3, instructions.Delay(11, d2))
+
+        self.assertEqual(schedule, reference)
+
+    def test_barrier_on_qubits(self):
+        """Test barrier directive on qubits."""
+        with pulse.build(self.backend) as schedule:
+            pulse.barrier(0, 1)
+
+        reference = pulse.Schedule()
+        reference += directives.RelativeBarrier(
```

Comment:
Is there a full barrier option, too?

Reply:
Oh, right, the barrier isn't for the user to interact with, right? What does "relative" mean here?

Reply:
A full barrier would be a `GlobalBarrier` and would need to unite separate schedules that exist in the same Qobj. I'm not sure of how to handle that in the builder right now. Relative means within this schedule only. I've inherited this naming from our internal syntax.

Reply:
Ah, by "full barrier option" I just meant, one across all qubits in this schedule. I'm confused now about the name though, what is it relative to?